identity/app/controllers/AccountDeletionController.scala

```diff
@@ -0,0 +1,162 @@
+        } yield ()).leftMap(_ => FailedToEnterCorrectPassword)
+
+      def unauthenticateUser(): EitherT[Future, AccountDeletionFailures, CookiesResponse] =
+        EitherT.fromEither(api.unauth(emailPasswdAuth, idRequest.trackingData)).leftMap(_ => FailedToUnauthenticateUser)
+
+      def removeFromAllMalingLists(): EitherT[Future, AccountDeletionFailures, ETSubscriber] =
+        EitherT(exactTargetService.unsubscribeFromAllLists(emailPasswdAuth.email)).leftMap(_ => FailedToRemoveFromAllEmailLists)
+
+      def deleteAccountProper(): EitherT[Future, AccountDeletionFailures, Unit] =
+        EitherT.fromEither(identityApiClient.deleteAccount(request.user.auth, emailPasswdAuth)).leftMap(_ => FailedToDeleteAccount)
+
+      def clearCookiesAndDisplaySuccessForm() =
+        NoCache(SeeOther(routes.AccountDeletionController.renderAccountDeletionConfirmForm().url))
+          .discardingCookies(cookiesToDiscard: _*)
+
+      def processFailures(failure: AccountDeletionFailures) = failure match {
+          case FailedToEnterCorrectPassword =>
+            SeeOther(routes.AccountDeletionController.renderAccountDeletionForm ().url)
+              .flashing(boundForm.withError ("password", "Password is incorrect").toFlash)
+
+          case FailedToUnauthenticateUser =>
+            logger.error(s"Failed to un-authenticate user ${request.user.user.id} during account deletion.")
+            renderFormWithUnableToDeleteAccountError(boundForm)
+
+          case FailedToRemoveFromAllEmailLists =>
+            logger.error(s"Failed to remove user ${request.user.user.id} from all mailing lists during account deletion.")
+            renderFormWithUnableToDeleteAccountError(boundForm)
+
+          case FailedToDeleteAccount =>
+            logger.error(s"Failed to delete account for user ${request.user.user.id}")
+            renderFormWithUnableToDeleteAccountError(boundForm)
+        }
+
+      (for {
+        _ <- checkUserEnteredCorrectPassword()
+        _ <- removeFromAllMalingLists()
+        _ <- unauthenticateUser()
+        _ <- deleteAccountProper()
+      } yield ()).fold(processFailures, _ => clearCookiesAndDisplaySuccessForm)
+    }
+
+    sealed trait DeletionType
+    case object AutoDeletion extends DeletionType
+    case class ManualDeletion(hasSavedArticles: Boolean, hasComment: Boolean, hasJob: Boolean, isSubscriber: Boolean, isMember: Boolean) extends DeletionType
+
+
+    /* Users can delete account themselves if they
+         - has no saved articles, and
+         - have never commented, and
+         - do not have a jobs account, and
+         - do not have active digipack subscriptions, and
+         - do not have active membership */
+    private def selectDeletionType[A](implicit request: AuthenticatedActions.AuthRequest[A]): Future[DeletionType] =
+      for {
+        hasSavedArticles <- identityApiClient.hasNonDefaultSavedArticles(request.user.auth)
```

Comment:
It doesn't look like there's much error handling going on here. What's the behaviour if a call fails?